thunder.post_update.php

```diff
@@ -22,3 +24,81 @@ function thunder_post_update_ensure_config_selector_installed() {
   // Output logged messages to related channel of update execution.
   return $thunderUpdater->logger()->output();
 }
+
+/**
+ * Installs the content_content_lock view.
+ */
+function thunder_post_update_add_content_lock_view() {
+
+  $content_view = View::load('content');
+  if (empty($content_view)) {
+    return;
+  }
+  $content_view->setThirdPartySetting('config_selector', 'feature', 'thunder_content_view');
+  $content_view->setThirdPartySetting('config_selector', 'priority', 0);
+
+  $view_contains_content_lock = FALSE;
+  foreach (array_keys($content_view->get('display')) as $display_name) {
+    $display = $content_view->getDisplay($display_name);
+    $options = ['fields', 'filters', 'sorts', 'relationships'];
+    foreach ($options as $option) {
+      if (!empty($display['display_options'][$option]) && is_array($display['display_options'][$option])) {
+        foreach ($display['display_options'][$option] as $key => $values) {
+          if ($values['table'] == 'content_lock') {
+            $view_contains_content_lock = TRUE;
+          }
+        }
+      }
+    }
+  }
+
+  if ($view_contains_content_lock) {
+    // Duplicate content into content_content_lock view and add third party
+    // settings.
+    $content_lock_view = $content_view->createDuplicate();
+    $content_lock_view->set('id', 'content_content_lock');
+    $content_lock_view->setThirdPartySetting('config_selector', 'feature', 'thunder_content_view');
+    $content_lock_view->setThirdPartySetting('config_selector', 'priority', 1);
+    $content_lock_view->save();
+
+    // Delete all content_lock related stuff from the content view.
+    foreach (array_keys($content_view->get('display')) as $display_name) {
+      $display = &$content_view->getDisplay($display_name);
+
+      // Delete content_lock relationships.
+      $deleted_relationships = [];
+      if (!empty($display['display_options']['relationships']) && is_array($display['display_options']['relationships'])) {
+        foreach ($display['display_options']['relationships'] as $key => $relationship) {
+          if ($relationship['table'] == 'content_lock') {
+            unset($display['display_options']['relationships'][$key]);
+            $deleted_relationships[] = $key;
+          }
+        }
+      }
+
+      // Delete content lock fields, filters and sorts and those that depend
+      // on a deleted relationship.
+      $options = ['fields', 'filters', 'sorts'];
+      foreach ($options as $option) {
+        if (!empty($display['display_options'][$option]) && is_array($display['display_options'][$option])) {
+          foreach ($display['display_options'][$option] as $key => $values) {
+            if ($values['table'] == 'content_lock' || (!empty($values['relationship']) && in_array($values['relationship'], $deleted_relationships))) {
+              unset($display['display_options'][$option][$key]);
+            }
+          }
+        }
+      }
+    }
+    $content_view->disable();
```

Comment:
Let's add a comment here that we're disabling with view without content lock. Because the content lock view is going to take precedence. Actually don't we want to import the non content lock view here? So they have a view to fallback to if they every uninstall content_lock? Atm this view will still depend on content_lock.

Reply:
Ah I see you are removing all the content_lock stuff above. Any reason why we don't just use the view that comes with Thunder?

Reply:
Exactly, we want that the view without the content_lock fields is very close to what had the user before.